osu.Game.Rulesets.Mania/Difficulty/Skills/README.md

```diff
@@ -0,0 +1,280 @@
+| 3    | 9  | 4   | 2   |      |    |    | Apply Decay                             |
+| 4    | 14 | 3   | 7   | 14+7 |    | X  | 3rd Note on Column 1, add to GS and CS1 |
+| 5    | 13 | 2   | 6   |      |    |    | Apply Decay                             |
+
+- GS increments on every note regardless of column, while CS only increments for their respective column.
+- S depends on GS and CS, never itself.
+    - On the 2nd note, S=GS+CS0 ignoring CS1 because the note's column is 0
+    - Same goes for the 3rd note, S=GS+CS1
+- CS is a matrix, GS is a vector.
+
+Though the actual script is non-trivial, it has similar concepts from the above example:
+
+- Our decay is [exponential](#evaluating-strains), so high strain values are exponentially rarer
+- We consider [Hold Interactions](#hold-strain-bonus-triggers), notes in complex Hold scenarios are have boosted
+  strains.
+- Decay is zero when evaluating notes at the same time (e.g. chords)
+- Strain is still the sum of Global and Column Strain
+
+### Strains
+
+We'll use math notations to define our strain robustly:
+
+- $GS_i$: GS on the $i^{th}$ object
+- $CS_{i,k}$: CS on the $i^{th}$ object on column $k$.
+- $S_i$: S on the $i^{th}$ object.
+- $O_i$: $i^{th}$ note/object
+- $\Delta_i$: time elapsed between the $i^{th}$ and previous object
+- $\Delta_{i,k}$: time elapsed between the $i^{th}$ and previous object in the same column $k$
+- $\alpha$: decay constant
+- $K$: Total keys
+- $N$: Total notes
+
+Where $GS$, $S$ are vectors, and $CS$ is a matrix of strains
+
+$$
+GS\in\mathbb R^N, CS\in\mathbb R^{N\times K}, S\in\mathbb R^N
+$$
+
+The 1st $GS_0$ is initialized with a value $GS_0=1$
+
+Subsequent $GS_i$ depends on
+
+- Previous value $GS_{i-1}$
+- Current and Past Notes $O_{i, i-1, ...}$
+- Strain Decay $Decay$
+
+$$GS_i=f(GS_{i-1}, O_{i, i-1, ...}, Decay(\Delta_i,\alpha_{GS}))$$
+
+The 1st $CS_0$ is initialized with $CS_0=(0,0,...,0)\in\mathbb R^K$
+
+Subsequent $CS_{i,k}$ values depend on
+
+$$CS_i=f(CS_{i-1}, O_{i, i-1, ...}, Decay(\Delta_{i,k},\alpha_{CS}))$$
+
+Finally, $S_i=GS_i+CS_{i,k}$ where $k$ is the note column
+
+#### Evaluating Strains
+
+To evaluate $GS_i$:
+
+1) We decay: $x=GS_{i-1}\times (\alpha_{GS})^{\Delta_i} : \alpha_{GS} < 1$
+2) Add bonus $B$ given the current and past notes: $GS_i=x+B|O_{i, i-1, ...}$
+
+Similarly for $CS_i$:
+
+1) We decay: $x=CS_{i-1}\times (\alpha_{CS})^{\Delta_{i,k}} : \alpha_{CS} < 1$
+2) Add bonus $B$ given the current and past notes: $CS_i=x+B|O_{i, i-1, ...}$
+
+> The bonuses are explained in the [Bonuses Evaluation](#bonuses-evaluation)
+
+---
+
+## Implementation
+
+### Rules
+
+Notes fed into `StrainValueOf` follow these rules:
+
+- Only 1 note is fed in at a time.
+- They are time sorted by the note head.
+- The first note is omitted.
+
+> - The 1st & 2nd rule implies that notes at the same time (e.g. in a chord) don't iterate into `StrainValueOf` deterministically.
```

Comment:
Explain why (it's because it makes values of global strains dependent on object - and therefore column - iteration order).